src/main/java/Drew.java

```diff
@@ -0,0 +1,309 @@
+
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import java.io.IOException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Main chatbot class. Contains the logic of the chatbot.
+ *
+ * @author cocoanautz
+ */
+public class Drew {
+    /**
+     * Checks the identity of the command.
+     *
+     * @param input String containing full user input.
+     * @return Command specified by user input.
+     */
+    public static Command checkCommandIdentity(String input) throws UnknownCommandException,
+            InsufficientArgumentsException, IllegalArgumentException {
+        int inputLength = input.length();
+        Command userCommand;
+        if (inputLength == 4 && input.substring(0, 4).equalsIgnoreCase("list")) {
+            userCommand = Command.LIST;
+        } else if (inputLength >= 4 && input.substring(0, 4).equalsIgnoreCase("mark")) {
+            userCommand = Command.MARK;
+        } else if (inputLength >= 6 && input.substring(0, 6).equalsIgnoreCase("unmark")) {
+            userCommand = Command.UNMARK;
+        } else if (inputLength >= 6 && input.substring(0, 6).equalsIgnoreCase("delete")) {
+            userCommand = Command.DELETE;
+        } else if (inputLength >= 4 && input.substring(0, 4).equalsIgnoreCase("todo")) {
+            userCommand = Command.TODO;
+        } else if (inputLength >= 8 && input.substring(0, 8).equalsIgnoreCase("deadline")) {
+            userCommand = Command.DEADLINE;
+        } else if (inputLength >= 5 && input.substring(0, 5).equalsIgnoreCase("event")) {
+            userCommand = Command.EVENT;
+        } else {
+            userCommand = Command.UNKNOWN;
+        }
+
+        String inputNoWhitespace = input.replaceAll("\\s", "");
+        int inputNoWhitespaceLength = inputNoWhitespace.length();
+        int backslashCount = 0;
+        for (int i = 0; i < inputNoWhitespaceLength; i++) {
+            char c = inputNoWhitespace.charAt(i);
+            if (c == '/') {
+                backslashCount++;
+            }
+        }
+        switch (userCommand) {
+        case UNKNOWN:
+            throw new UnknownCommandException("Command not recognized");
+        case LIST:
+            if (inputNoWhitespaceLength != 4) {
+                throw new IllegalArgumentException("No arguments are needed");
+            }
+            break;
+        case TODO:
+            if (inputNoWhitespaceLength == 4) {
+                throw new InsufficientArgumentsException("'Todo task' cannot be empty");
+            }
+            break;
+        case MARK:
+            if (inputNoWhitespaceLength == 4) {
+                throw new InsufficientArgumentsException("'Mark index' cannot be empty");
+            }
+            break;
+        //valid integer will be checked later
```

Comment:
Comments should target other programmers reading the code and not only make sense to the person who wrote it